source/DamageProfile.cpp

```diff
@@ -111,11 +111,14 @@ void DamageProfile::PopulateDamage(DamageDealt &damage, const Ship &ship) const
 	double shieldFraction = 0.;
 
 	// Lambda for returning the damage scale that a damage type should
-	// use given the default percentage that is blocked by shields and
-	// the value of its protection attribute.
-	auto ScaleType = [&](double blocked, double protection)
+	// use given the default percentage that is blocked by shields and hull,
+	// and the value of its protection attribute.
+	auto ScaleType = [&](double shieldBlocked, double hullBlocked, double protection)
 	{
-		return damage.scaling * (1. - blocked * shieldFraction) / (1. + protection);
+		return damage.scaling *
+			(((1. - shieldBlocked) * (shieldFraction))
+			+ ((1. - hullBlocked) * (1. - shieldFraction)))
+			/ (1. + protection);
```

Comment:
Played around with this a bit more after the fix, and I think it checks out now. I do think we should bust it out into its own line for readability, though.
```suggestion
		double blocked = (1. - shieldBlocked) * (shieldFraction) + (1. - hullBlocked) * (1. - shieldFraction);
		return damage.scaling * blocked / (1. + protection);
```